### PR TITLE
Introduce ExternalError and output external errors with a new component

### PR DESCRIPTION
**Pull request by matteodepalo**
### WHY are these changes introduced?

@isaacroldan reported that he saw an error that broke the borders of the banner component.

Captura de Pantalla 2022-10-20 a las 12 44 31

### WHAT is this pull request doing?

I've introduced a new type of error called `ExternalError`. This is used when rescuing errors from `execa` and allows us to print these kind of errors differently, like so:

Screenshot 2022-11-01 at 13 49 19

### Measuring impact

How do we know this change was effective? Please choose one:

- [x] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix
- [ ] Existing analytics will cater for this addition
- [ ] PR includes analytics changes to measure impact

### Checklist

- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible [documentation](https://shopify.dev) changes
- [x] I've made sure that any changes to `dev` or `deploy` have been reflected in the [internal flowchart](https://www.figma.com/file/7vqUp50u6dm48Zfb4JRRn8/CLI3-Internals).

### CODE REVIEW

**Comment by isaacroldan** [comment]
Fast! ⚡ 
Were you able to reproduce a similar error without the box bug?

**Comment by matteodepalo** [comment]
Unfortunately no :/ I couldn't find an error that would have that behavior. Did you run some specific command?

**Comment by isaacroldan** [comment]
Yeah, this was on a specific project that didn't support node18, so the `yarn install` we execute inside `dev` fails.
I'll try to tophat this today 🙏

**Comment by isaacroldan** [comment]
Captura de Pantalla 2022-10-21 a las 13 54 32
Works 👌

**Review by isaacroldan — APPROVED** [review]

File: packages/cli-kit/src/private/node/ui/components/FatalError.tsx [hunk 1]
```diff
@@ -4,6 +4,7 @@ import {Bug, cleanSingleStackTracePath, Fatal} from '../../../../error.js'
 import {Box, Text} from 'ink'
 import React from 'react'
 import StackTracey from 'stacktracey'
+import stripAnsi from 'strip-ansi'
```
Comment (pepicrft) [inline packages/cli-kit/src/private/node/ui/components/FatalError.tsx:7]: I'd add a changeset update to mention the fix.

**Comment by matteodepalo** [comment]
@amcaplan wow that link is really interesting! Definitely something to consider in the future. If we want to respect 3P output properly we need something like that in place. For now though I'd rather merge this PR, but I'll write a note for that. I also need to make a spike about 3P errors in general so it will come useful soon.

**Comment by matteodepalo** [comment]
I don't think we can be that surgical and remove only the characters to preserve indentations. I would go with a flat line at the top and everything that comes from external tools aligned to the left as it's the approach that has less risk.

**Comment by matteodepalo** [comment]
Screenshot 2022-11-01 at 13 49 19
@MeredithCastile I've just made the above change in this PR. How does that look? We can say "error coming from _command that was run_".

**Comment by matteodepalo** [comment]
@isaacroldan @pepicrft could I get another review? The code has changed quite a lot.

**Review by isaacroldan — APPROVED** [review]

**Pull request merged**